### PR TITLE
Issue198: add project participant status to projects

### LINKED ISSUE #198: add project participant status
add a status field for each participant, with three statuses available to all ols-2 participants: active, inactive, or graduated

### PR DESCRIPTION
**Pull request by vasconsaurus**
closes #198 

### **What this PR does**
This adds a `status` field for each project's participant.

### **How/Why**
I added the `status` inside an `if clause` to the `projects-participants.md`. I did this because we don't have all the status** data yet. This way the `status` will only show if it exists inside the `ols-x-projects.yaml`.

If this is ok, we could update the structure with this PR, and I can open a new issue to update the YAML files when the data is available.

** about the the status data: As @AndreaSanchezTapia pointed out, I found the `graduated:true` field, but only in the `ols-1-projects.yaml`

### **Some images of the new behaviour 😉**
without status
Screen Shot 2022-10-22 at 12 32 39

with status
Screen Shot 2022-10-22 at 12 32 20

### CODE REVIEW

**Comment by vasconsaurus** [comment]
@bebatut and @AndreaSanchezTapia could you please review this one for me? ❤️

**Comment by vasconsaurus** [comment]
@bebatut, if these changes are accepted, I was thinking that maybe we should create a new issue to remove the `graduated: true` field and update the places where it is used to use `status: graduated`.

This way we avoid the duplication of this data, and I think the new `status` data is more complete.  I could do this as a follow-up to this work. What do you think?

File: _ols-1/projects-participants.md [hunk 1]
```diff
@@ -42,12 +42,18 @@ For the first round of the Open Life Science program, we are happy to have [{{ p
         {% endfor %}
         {% endcapture %}
 
+        {% assign status = project.status %}
```
Comment (bebatut) [inline _ols-1/projects-participants.md:45]: I do not think you really need the `assign` here

File: _ols-1/projects-participants.md [hunk 2]
```diff
@@ -42,12 +42,18 @@ For the first round of the Open Life Science program, we are happy to have [{{ p
...
+
 ## {{ project.name }}
 
 **By**: {{ p-pparticipants | remove_first: ', ' }}
 
 **Mentored by**: {{ p-mentors | remove_first: 'with ' }}
 
+{% if project.status %}
+**Status**: {{ status }}
```
Comment (bebatut) [inline _ols-1/projects-participants.md:54]:
```suggestion
**Status**: {{ project.status }}
```

**Review by bebatut — COMMENTED** [review]
The changes look good. 

I just merged the PR #490 from @AndreaSanchezTapia.
Could you merge the main branch here and then add the status also to the project files for the other cohorts?

Regarding replacing `graduated` by `status`, I agree we should keep only `status`.
Could you make that directly in this PR so it is complete?

Thanks

**Comment by vasconsaurus** [comment]
> The changes look good.
> 
> I just merged the PR #490 from @AndreaSanchezTapia. Could you merge the main branch here and then add the status also to the project files for the other cohorts?
> 
> Regarding replacing `graduated` by `status`, I agree we should keep only `status`. Could you make that directly in this PR so it is complete?
> 
> Thanks

will do! 🙂

**Comment by vasconsaurus** [comment]
> I just merged the PR #490 from @AndreaSanchezTapia. Could you merge the main branch here and then add the status also to the project files for the other cohorts?
> 

@bebatut I'm using gitpod, I tried `git pull `/ `git pull origin main`, but it says my main is up to date, even though it isn't. Do you have any idea why this could be?

**Comment by bebatut** [comment]
Can you check with `git remote -v` if origin is there or your fork?

**Comment by vasconsaurus** [comment]
> Can you check with `git remote -v` if origin is there or your fork?

aaaaaa! I had forked it and forgot! 🙈 I will fix this, but while I was waiting I installed it locally, so I might work locally now 

thanks @bebatut ❤️

**Comment by vasconsaurus** [comment]
@bebatut I finished updating, could you see if this is ok?

And just to note, currently, we only have `status: graduated` and don't have any `active` or `inactive`. But I'm guessing we'll have those as the new cohort begins?

**Comment by vasconsaurus** [comment]
updated as per this [comment](https://github.com/open-life-science/open-life-science.github.io/pull/490#issuecomment-1288997514)

**Comment by bebatut** [comment]
> @bebatut I finished updating, could you see if this is ok?
> 
> And just to note, currently, we only have `status: graduated` and don't have any `active` or `inactive`. But I'm guessing we'll have those as the new cohort begins?

Yes it will come with the new cohort

File: Gemfile.lock [hunk 3]
```diff
@@ -221,6 +221,8 @@ GEM
       jekyll-feed (~> 0.9)
       jekyll-seo-tag (~> 2.1)
     minitest (5.16.1)
+    nokogiri (1.13.9-arm64-darwin)
```
Comment (bebatut) [inline Gemfile.lock:224]: Could you please revert your changes on this file? Thanks

File: _ols-1/projects-participants.md [hunk 4]
```diff
@@ -13,10 +13,10 @@ photos:
 
 {% assign all-participants = '' %}
 {% for project in projects %}
-    {% assign p-pparticipants = '' %}
-    {% for p in project.participants %}
-        {% capture all-participants %}{{ all-participants}}, {{ p }}{% endcapture %}
-    {% endfor %}
+{% assign p-pparticipants = '' %}
```
[the remark this answers is not in the document]
Comment (vasconsaurus) [reply _ols-1/projects-participants.md:16]: yes! I had some problems with the vscode removing indentation when I saved `.md` files, I solved it but missed this one.

File: Gemfile.lock (hunk 3 above)
Comment (vasconsaurus) [reply Gemfile.lock:224]: The `gemfile.lock` was generated by the `gemfile` when I ran `bundle`, I didn't edit anything in the original `gemfile`. I also ran `gem install jekyll bundler` to install jekyll, maybe those came from there and are jekyll dependencies?

I think the `arm64` is only support for arm architecture.

@bebatut what would be the best way to go about this?

File: _ols-1/projects-participants.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (vasconsaurus) [reply _ols-1/projects-participants.md]: well, vscode tried to sabotage me 😅, but I think I got all of them

File: Gemfile.lock (hunk 3 above)
Comment (bebatut) [reply Gemfile.lock:224]: could you revert your changes on this file and make sure you do not add this file to the commits?

Comment (vasconsaurus) [reply Gemfile.lock:224]: @bebatut it adds those every time I run the jekyll server

so if I needed to see something I would need to start the server, stop it, revert the changes, make my changes, add/commit, start again / stop / revert etc...

are those jekyll dependencies a problem?

I could work from gitlab, but that has a time limit for the free tier.. so I don't think that is much of a solution 🙁

**Comment by AndreaSanchezTapia** [comment]
You can avoid committing that file, add only the ones you are working on.

**Comment by vasconsaurus** [comment]
> You can avoid committing that file, add only the ones you are working on. 

....... of course ......... I am so dumb 🙃 

but I think I'll have to do it from the beginning since that is the very first commit, right?

**Comment by vasconsaurus** [comment]
@bebatut The commits history got all messy because I had to start from scratch, BUT I checked which files were changed, and now it seems we are good to go 🙂 (fingers crossed!)

**Review by bebatut — APPROVED** [review]
Looks good now. Thanks @vasconsaurus

**Pull request merged**